addons/web/views/webclient_templates.xml

```diff
@@ -463,6 +463,12 @@
         <xpath expr="//head/meta[last()]" position="after">
             <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>
         </xpath>
+        <xpath expr="//head" position="inside">
+            <script type="text/javascript">
+                const userTZ = Intl.DateTimeFormat().resolvedOptions().timeZone;
+                document.cookie = `tz=${userTZ}`;
```

Comment:
Do we perhaps want to put the cookie on `/` at all times, to avoid generating multiple cookies after visiting different URLs on the website (similarly to the way our `utils.set_cookie()` forces the path)? Currently if you visit `/` then `/web` and you get 2 copies of the cookie, as I think browsers consider this a different cookie.

Unrelated: the default "session" expiration sounds appropriate :+1: 

```suggestion
                document.cookie = `tz=${userTZ}; path=/`;
```